### PR TITLE
⚡  perf: cache permissions, simplify user caching

### PR DESCRIPTION
**Pull request by larwaa**
## Proposed Changes

- Cache permissions ⚡ 
- Remove custom caching logic for users, enabling Apollo to handle it automagically 🧙 
- Add dynamic imports for `FirstLogin` and `EditUser` dialogs

## Types of Changes

- [ ] New feature
- [ ] Bug fix
- [x] Enhancement/optimization
- [ ] Refactor
- [ ] Style
- [ ] Build/dependencies
- [ ] Other

## Issue(s) fixed or closed by this PR

- Flashing "Archive"-nav item for all Indøk-users.

## Checklist

- [ ] I have added tests to cover my changes (for features/bugs)
- [x] I am pleased with the readability of the code (if not, state where you need input)
- [x] I have tested the changes and verified that they work and do not break anything (to the best of my ability)

### CODE REVIEW

File: frontend/src/graphql/users/mutations.ts [hunk 1]
```diff
@@ -1,24 +1,13 @@
 import { gql } from "@apollo/client";
+import { USER_FRAMGENT } from "./fragments";
 
 export const AUTHENTICATE = gql`
+  ${USER_FRAMGENT}
   mutation AuthUser($code: String!) {
     authUser(code: $code) {
-      token
```
Comment (larwaa) [inline frontend/src/graphql/users/mutations.ts:6]: Unused

File: frontend/src/pages/authCallback.tsx [hunk 2]
```diff
@@ -10,28 +10,22 @@ import Link from "next/link";
...
-      authUser({
-        variables: { code },
-        update: (cache, { data }) => {
-          if (!data || !data.authUser || !data.authUser.user) {
-            return;
-          }
-          cache.writeQuery<User>({ query: GET_USER, data: data.authUser.user });
-        },
-      });
-    }
```
Comment (larwaa) [inline frontend/src/pages/authCallback.tsx:34]: Trying to write `authUser` to the `GET_USER` query fails as Apollo expects `organizations: Organization[]` and `events: Event[]` fields. Thus, a refetch will occur anyway on the next page load.

File: frontend/src/components/pages/profile/EditUser.tsx [hunk 3]
```diff
@@ -77,14 +76,7 @@ export const EditUser: React.FC<EditUserProps> = ({ open, user, onSubmit, onClos
...
     updateUser({
       variables: { userData: updateUserInput },
-      update: (cache, { data }) => {
-        if (!data || !data.updateUser || !data.updateUser.user) {
-          return;
-        }
-        cache.writeQuery<User>({ query: GET_USER, data: data.updateUser.user });
-        onSubmit();
-      },
-    });
```
Comment (larwaa) [inline frontend/src/components/pages/profile/EditUser.tsx:87]: These consistently raised errors due to missing fields. Apollo resolves this automagically with [Store normalized objects](https://www.apollographql.com/docs/react/caching/overview/#4-store-normalized-objects) 🧙‍♀️

File: frontend/src/interfaces/users.ts [hunk 4]
```diff
@@ -1,6 +1,7 @@
 import { Organization } from "./organizations";
 
 export interface User {
```
Comment (MorganFH) [inline frontend/src/interfaces/users.ts:3]:
```suggestion
export type User = {
```

**Review by MorganFH — DISMISSED** [review]
LGTM

**Review by MorganFH — APPROVED** [review]
LGTM

**Pull request merged**